.github/workflows/push.yml

```diff
@@ -46,6 +46,10 @@ jobs:
         shell: bash
         run: poetry config virtualenvs.in-project true
 
+      - name: Check poetry lock
```

Comment:
The workflow here would still pass if the lock file is out of date, but the dependencies are otherwise installed correctly (the updated lock file would not then be committed back to the repo).

You may want `poetry check --lock' instead which just compares against the pyproject file and will exit with an error and halt the workflow if they do not match up.

Reply:
Good shout!

Reply:
Ah, think it'll be `poetry lock --check`. `poetry check` in general I don't get, the docs are light on when and why it will fail - I was testing it and it didn't raise when I added a totally bogus dependency to `pyproject.toml` that obviously wasn't reflected in the lock file.

Reply:
I get a deprecation warning when I try that.

`poetry lock --check is deprecated, use `poetry check --lock` instead`

Reply:
`poetry check --lock` seems to have the desired behaviour when I try it.